vnext/Microsoft.ReactNative/IReactNotificationService.h

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "ReactNotificationServiceHelper.g.h"
+#include <functional/functorRef.h>
+#include <object/objectRefCount.h>
+#include <winrt/Microsoft.ReactNative.h>
+#include <atomic>
+#include <map>
+#include <mutex>
+#include <vector>
+
+namespace winrt::Microsoft::ReactNative::implementation {
+
+struct ReactNotificationArgs : implements<ReactNotificationArgs, IReactNotificationArgs> {
+  ReactNotificationArgs(IReactNotificationSubscription const &subscription, IInspectable const &data) noexcept
+      : m_subscription{subscription}, m_data{data} {}
+
+  IReactNotificationSubscription Subscription() const noexcept {
+    return m_subscription;
+  }
+
+  IInspectable Data() const noexcept {
+    return m_data;
+  }
+
+ private:
+  IReactNotificationSubscription m_subscription;
+  IInspectable m_data;
+};
+
+struct ReactNotificationService : implements<ReactNotificationService, IReactNotificationService> {
+  ReactNotificationService();
+  explicit ReactNotificationService(IReactNotificationService const parentNotificationService) noexcept;
+  ~ReactNotificationService() noexcept;
+
+  IReactNotificationSubscription Subscribe(
```

Comment:
>Subscribe [](start = 33, length = 9)

Just curious, we have CallbackStore for events in devmain, would it make sense to use something like that here?

Reply:
This implementation uses a lot of ideas from the CallbackStore. But, it has some more advanced features such as the IReactSubscription object. It is much more powerful when retuning just a 'cookie'

---
In reply to: [427618875](https://github.com/microsoft/react-native-windows/pull/4953#discussion_r427618875) [](ancestors = 427618875)